synapse/handlers/sync.py

```diff
@@ -485,22 +495,60 @@ def compute_state_delta(self, room_id, batch, sync_config, since_token, now_toke
         # TODO(mjark) Check for new redactions in the state events.
 
         with Measure(self.clock, "compute_state_delta"):
+
+            types = None
+            member_state_ids = {}
+            lazy_load_members = sync_config.filter_collection.lazy_load_members()
+
+            if lazy_load_members:
+                # We only request state for the members needed to display the
+                # timeline:
+
+                types = [
+                    (EventTypes.Member, state_key)
+                    for state_key in set(
+                        event.sender  # FIXME: we also care about invite targets etc.
+                        for event in batch.events
+                    )
+                ]
+
+                # We can't remove redundant member types at this stage as it has
```

Comment:
what are member types, and why would they be redundant?

Reply:
this was a note to self which snuck in.